src/dmd/backend/dlist.d

```diff
@@ -0,0 +1,570 @@
+/**
+ * Interface to the C linked list type.
+ *
+ * List is a complete package of functions to deal with singly linked
+ * lists of pointers or integers.
+ * Features:
+ *      1. Uses mem package.
+ *      2. Has loop-back tests.
+ *      3. Each item in the list can have multiple predecessors, enabling
+ *         different lists to 'share' a common tail.
+ *
+ * Copyright:   Copyright (C) 1986-1990 by Northwest Software
+ *              Copyright (c) 1999-2016 by Digital Mars, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(DMDSRC backend/tk/_dlist.d)
+ */
+
+module dmd.backend.dlist;
+
+import core.stdc.stdarg;
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+extern (C++):
+
+nothrow @nogc
+{
+
+/* **************** TYPEDEFS AND DEFINES ****************** */
+
+struct LIST
+{
+        /* Do not access items in this struct directly, use the         */
+        /* functions designed for that purpose.                         */
```

Comment:
Can you make them private now?

Reply:
Gawd will smite me if I violate the #1 rule of translation.

Reply:
Note that if I make it private now, then I have to add to this PR all the other files that are modified to make that work. Of course, that will inevitably cause more problems. Such changes need to be separate PRs.